### PR TITLE
Simplifier pass PR

### PR DESCRIPTION
**Pull request by certik**
GitHub made a mistake and changed the branch in #4700, and there does not seem to be a way to fix it, so I am opening a new PR.

### CODE REVIEW

**Comment by certik** [comment]
The conflicts are resolved in https://github.com/lfortran/lfortran/pull/5181.

**Comment by gxyd** [comment]
Seems like one current failure is a real one.

**Comment by gxyd** [comment]
Did someone restart the job here? I didn't push anything, and the tests pass now

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -805,9 +805,7 @@ jobs:
             ctest --output-on-failure
             cd integration_tests
             ./run_tests.py -b llvm llvmImplicit
-            ./run_tests.py -b llvm llvmImplicit -f
-            ./run_tests.py -b llvm llvmImplicit --experimental-simplifier
-            ./run_tests.py -b llvm llvmImplicit -f --experimental-simplifier
+            #./run_tests.py -b llvm llvmImplicit -f
```
Comment (gxyd) [inline .github/workflows/CI.yml:808]: @certik , just that this doesn't come as a total surprise to you that this is currently disabled. I'll come to it to see what fails with this.

**Comment by certik** [comment]
Closing in favor of https://github.com/lfortran/lfortran/pull/5198. Same exact branch, just my personal fork instead.

**Pull request closed without merging**